dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/Responses/Converters/AfItemResourceGenerator.cs

```diff
@@ -0,0 +1,306 @@
+            {
+                SequenceNumber = this.Seq.GetNext(),
+                ItemId = item.Id,
+                OutputIndex = this.GroupSeq.Current(),
+                Arguments = item.Arguments
+            };
+
+            yield return new StreamingOutputItemDone
+            {
+                SequenceNumber = this.Seq.GetNext(),
+                OutputIndex = groupSeq,
+                Item = item
+            };
+        }
+    }
+
+    private async IAsyncEnumerable<StreamingResponseEvent> GenerateFunctionCallOutputEventsAsync(
+        IAsyncEnumerable<AIContent> source,
+        Action<ItemResource> onItemResource,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        await foreach (var content in source.WithCancellation(cancellationToken).ConfigureAwait(false))
+        {
+            if (content is not FunctionResultContent functionResultContent)
+            {
+                continue;
+            }
+
+            var groupSeq = this.GroupSeq.GetNext();
+            var item = functionResultContent.ToFunctionToolCallOutputItemResource(this.Context.IdGenerator.GenerateFunctionOutputId());
+            onItemResource(item);
+
+            yield return new StreamingOutputItemAdded
+            {
+                SequenceNumber = this.Seq.GetNext(),
+                OutputIndex = groupSeq,
+                Item = item
+            };
+
+            yield return new StreamingOutputItemDone
+            {
+                SequenceNumber = this.Seq.GetNext(),
+                OutputIndex = groupSeq,
+                Item = item
+            };
+        }
+    }
+
+    private async IAsyncEnumerable<StreamingResponseEvent> GenerateAssistantMessageEventsAsync(
+        IAsyncEnumerable<AIContent> source,
+        Action<ItemResource> onItemResource,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var groupSeq = this.GroupSeq.GetNext();
+        var itemId = this.Context.IdGenerator.GenerateMessageId();
+        var incompleteItem = new ResponsesAssistantMessageItemResource(
+            id: itemId,
+            status: ResponsesMessageItemResourceStatus.InProgress,
+            content: []
+        );
+
+        yield return new StreamingOutputItemAdded
+        {
+            SequenceNumber = this.Seq.GetNext(),
+            OutputIndex = groupSeq,
+            Item = incompleteItem
+        };
+
+        yield return new StreamingContentPartAdded
+        {
+            SequenceNumber = this.Seq.GetNext(),
+            ItemId = itemId,
+            OutputIndex = groupSeq,
+            ContentIndex = 0,
+            Part = new ItemContentOutputText(string.Empty, [])
+        };
+
+        var text = new StringBuilder();
+        await foreach (var content in source.WithCancellation(cancellationToken).ConfigureAwait(false))
+        {
+            if (content is not TextContent textContent)
```

Comment:
It looks like we're not surfacing other forms of content, like reasoning (TextReasoningContent)? We can of course add it later.